src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -1144,13 +1144,25 @@ SystemNative_SetIPv6Address(uint8_t* socketAddress, int32_t socketAddressLen, ui
     return PAL_SUCCESS;
 }
 
-static void ConvertMessageHeaderToMsghdr(msghdr* header, const MessageHeader& messageHeader)
+static bool IsStreamSocket(int socket)
+{
+    int type;
+    socklen_t length = sizeof(int);
+    return socket != -1
+            && getsockopt(socket, SOL_SOCKET, SO_TYPE, &type, &length) == 0
+            && type == SOCK_STREAM;
```

Comment:
Is there a reasonable way we could add a test for the SOCK_DGRAM case?

Reply:
We can verify sends/receives are not being truncated. I'll add a test for this.